tests/jax/components/executing/observing_test.py

```diff
@@ -0,0 +1,286 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for FeedforwardExecutorObserve class for Jax-based Mava systems"""
+
+
+from types import SimpleNamespace
+from typing import Dict, Tuple
+
+import numpy as np
+import pytest
+from acme.types import NestedArray
+from dm_env import StepType, TimeStep
+
+from mava.components.jax.executing.observing import (
+    ExecutorObserveConfig,
+    FeedforwardExecutorObserve,
+)
+from mava.systems.jax.executor import Executor
+from mava.types import OLT
+from mava.utils.sort_utils import sort_str_num
+
+
+class MockAdder:
+    def __init__(self) -> None:
+        pass
+
+    def add_first(
+        self, timestep: TimeStep, extras: Dict[str, NestedArray] = ...  # type: ignore
+    ) -> None:
+        self.test_timestep = timestep
+        self.test_extras = extras
+
+    def add(
+        self,
+        actions: Dict[str, NestedArray],
+        next_timestep: TimeStep,
+        next_extras: Dict[str, NestedArray] = ...,  # type: ignore
+    ) -> None:
+        self.test_adder_actions = actions
+        self.test_next_timestep = next_timestep
+        self.test_next_extras = next_extras
+
+
+class MockExecutorParameterClient:
+    def __init__(self) -> None:
+        self.parm = False
+
+    def get_async(self) -> None:
+        self.parm = True
+
+
+@pytest.fixture
+def executor_without_adder() -> Executor:
+    """Mock executor component without adder"""
+    extras = {
+        "agent_0": np.array([0]),
+        "agent_1": np.array([1]),
+        "agent_2": np.array([2]),
+    }
+    store = SimpleNamespace(
+        is_evaluator=None, observations={}, adder=None, extras=extras
+    )
+    return Executor(store=store)
+
+
+class MockExecutor(Executor):
+    def __init__(self, *args: object) -> None:
+        # agent_net_keys
+        agent_net_keys = {
+            "agent_0": "network_agent_0",
+            "agent_1": "network_agent_1",
+            "agent_2": "network_agent_2",
+        }
+        # network_int_keys_extras
+        network_int_keys_extras = None
+        # network_sampling_setup
+        network_sampling_setup = [
+            ["network_agent_0"],
+            ["network_agent_1"],
+            ["network_agent_2"],
+        ]
```

Comment:
Just a heads up here, agent networks will now be randomly sampled. Is this intended?

Reply:
Yeah it is if we use    ` [ ["network_agent_0","network_agent_1","network_agent_2"]] `there won't be shuffle when we use the `sample_new_agent_keys` function it will return same network however if we use the above we will get random values between 0 and 2. Is this right or might I be mistaken?

Reply:
You are correct, I just wanted to double check that you wanted to be doing that :slightly_smiling_face: